### PR TITLE
[ci] Small improvements for run-jenkins and collect-coverage.

### PR DESCRIPTION
**Pull request by vkargov**

### CODE REVIEW

File: scripts/ci/collect-coverage.sh [hunk 1]
```diff
@@ -3,18 +3,19 @@
 # This script is meant to be executed on all "slave" machines that run coverage collection.
 
 COV_DIR=coverage
-COV_INFO="$COV_DIR/$JOB_NAME.info"
+COV_NAME=`echo $JOB_NAME | sed 's#/#-#g'`.info
```
Comment (akoeplinger) [inline scripts/ci/collect-coverage.sh]: I prefer the `$()` syntax over backticks

File: scripts/ci/collect-coverage.sh [hunk 2]
```diff
@@ -3,18 +3,19 @@
...
+COV_INFO="$COV_DIR/$COV_NAME"
 
 # Build Mono and collect coverage on the test suite.
-CI_TAGS=collect-coverage,monolite scripts/ci/run-jenkins.sh
+CI_TAGS=collect-coverage,monolite,"$CI_TAGS" scripts/ci/run-jenkins.sh
```
Comment (akoeplinger) [inline scripts/ci/collect-coverage.sh]: please quote the whole arg instead: `CI_TAGS="collect-coverage,monolite,$CI_TAGS"`

File: scripts/ci/run-jenkins.sh [hunk 3]
```diff
@@ -108,6 +106,8 @@ ${TESTCMD} --label=make --timeout=${make_timeout} --fatal make ${make_parallelis
 if [[ ${CI_TAGS} == *'checked-coop'* ]]; then export MONO_CHECK_MODE=gc,thread; fi
 if [[ ${CI_TAGS} == *'checked-all'* ]]; then export MONO_CHECK_MODE=all; fi
 
+export MONO_ENV_OPTIONS="$MONO_ENV_OPTIONS $MONO_TEST_FLAGS"
```
Comment (akoeplinger) [inline scripts/ci/run-jenkins.sh]: why do we need another variable? shouldn't MONO_ENV_OPTIONS be enough?

File: scripts/ci/collect-coverage.sh (hunk 2 above)
Comment (vkargov) [reply scripts/ci/collect-coverage.sh]: OK, I'll change that, though these 2 forms should are absolutely identical.

File: scripts/ci/run-jenkins.sh (hunk 3 above)
Comment (vkargov) [reply scripts/ci/run-jenkins.sh]: MONO_ENV_OPTIONS will affect the Mono build while MONO_TEST_FLAGS should only  affect the tests. That may make sense if you want to use the default options while building mscorlib, but pass some exotic option to the test suite. As one example, building mscorlib seems to fail with MONO_ENV_OPTIONS=--llvm, even though --llvm does work on most benchmarks.

Comment (akoeplinger) [reply scripts/ci/run-jenkins.sh]: OK, makes sense. I'd call it `MONO_TEST_ENV_OPTIONS` though so it's more consistent.

**Comment by vkargov** [comment]
Updated the PR.

**Pull request merged**